### PR TITLE
net-misc/zerotier: add 1.10.6

### PR DESCRIPTION
**Pull request by parona-source**
Closes: https://bugs.gentoo.org/884897

Ill note that Ive only tested amd64. And that SSO is a paid feature which I havent tested at runtime, it compiles correctly though. Its built as a static lib and then linked with zerotier-one binary.

### CODE REVIEW

**Comment by parona-source** [comment]
Typos and give the debug use flag justification to exists even though it would just add preprocessor arguments when sso is disabled.

**Comment by parona-source** [comment]
@thesamesam ill just ping here as well. no maintainer, but a more substantial bump.

I get that debug use flag may be suspect when you arent building the optional rust bit. but having it makes it simpler overall. doing REQUIRED_USE="debug? ( sso )" is imo unnecessary.

File: net-misc/zerotier/zerotier-1.10.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+IUSE="cpu_flags_arm_neon debug sso"
+
+RDEPEND="
+	dev-libs/openssl:=
+	net-libs/libnatpmp
+	>=net-libs/miniupnpc-2:=
+"
+DEPEND="${RDEPEND}"
+
+BDEPEPED="
```
Comment (thesamesam) [inline net-misc/zerotier/zerotier-1.10.6.ebuild:219]: 👀

File: net-misc/zerotier/zerotier-1.10.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+DOCS=( README.md AUTHORS.md )
+
+src_unpack() {
+	unpack ${P}.tar.gz
+	use sso && cargo_src_unpack
+}
+
+src_prepare() {
+	default_src_prepare
```
Comment (thesamesam) [inline net-misc/zerotier/zerotier-1.10.6.ebuild:238]: Let's do just default

File: net-misc/zerotier/zerotier-1.10.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+		ZT_DEBUG="$(usex debug 1 0)"
+		ZT_SSO_SUPPORTED="$(usex sso 1 0)"
+	)
+
+	pushd zeroidc > /dev/null || die
+	use sso && cargo_src_compile
+	popd > /dev/null || die
+
+	emake "${myemakeargs[@]}" one
+
```
Comment (thesamesam) [inline net-misc/zerotier/zerotier-1.10.6.ebuild:278]: Drop blank line at end

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: net-misc/zerotier/zerotier-1.10.6.ebuild (hunk 1 above)
Comment (parona-source) [reply net-misc/zerotier/zerotier-1.10.6.ebuild:218]: I hope understood correctly that it was just the stray newline rather something else like spacing with the BDEPEND use or the rust dep

Comment (thesamesam) [reply net-misc/zerotier/zerotier-1.10.6.ebuild:218]: No, typo in the variable name.

Comment (parona-source) [reply net-misc/zerotier/zerotier-1.10.6.ebuild]: Truly a blind spot. Looked at everything around it and not at it. Thank you pointing that one, quite embarrassing

Comment (thesamesam) [reply net-misc/zerotier/zerotier-1.10.6.ebuild]: It happens :)

**Pull request closed without merging**